### DISCUSSION TITLE
How to forward multiple users to the same SMTP relay server?

### DISCUSSION THREAD

**Discussion by CBAnother** [docker-mailserver/docker-mailserver#4028] [Q&A] [answered]
Hello

My cloud service provider does not open SMTP or ESMTP ports and I can only use their provided SMTP relay service. I have multiple users who need mailing services, and my configuration method is as follows, edit `compose.yaml`

```
volumes:
- ./postfix-main.cf:/tmp/docker-mailserver/postfix-main.cf
- ./sasl_passwd:/tmp/docker-mailserver/sasl_passwd
- ./sender_relay:/tmp/docker-mailserver/sender_relay
```

edit `postfix-main.cf`

```
smtp_sasl_password_maps = texthash:/tmp/docker-mailserver/sasl_passwd
sender_dependent_relayhost_maps = texthash:/tmp/docker-mailserver/sender_relay
smtp_sasl_auth_enable = yes
smtp_sasl_security_options = noanonymous
smtp_tls_wrappermode = yes
smtp_tls_security_level = encrypt
```

edit `sasl_passwd`

```
user1@xxx.fit user1@xxx.fit:smtp_relay_password1
user2@xxx.fit user1@xxx.fit:smtp_relay_password2
```

edit `sender_relay`

```
user1@xxx.fit    [xxx.smtp-relay.com]:465
user2@xxx.fit    [xxx.smtp-relay.com]:465
```

But the sending of the email failed, and the log results are as follows.

```
postfix/smtp[3824]: 66CD81A198D: to=<xxx@xxxx.com>, relay=xxx.smtp-relay.com[a.b.c.d]:465, delay=0.18, delays=0.01/0.01/0.13/0.03, dsn=5.0.0, status=bounced (host xxx.smtp-relay.com[a.b.c.d] said: 531 Authentication is required [@sm050105] (in reply to MAIL FROM command))
```

I modified `sasl_passwd` as follows, `user1` can send successfully, but since the first field of this file cannot be repeated, only `user1` can succeed.

```
[xxx.smtp-relay.com]:465 user1@xxx.fit:smtp_relay_password1
user2@xxx.fit user1@xxx.fit:smtp_relay_password2
```

What should I do? I guess it may be a switch that I did not toggle, but I did not find it when I looked up the documentation.

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Hello,

Based on the information you provided and the configuration you are using, it seems like the issue is related to how the `sasl_passwd` file is being utilized. In your initial setup, the `sasl_passwd` file entries were not correctly formatted to match the relay host and port with the corresponding user credentials. This is why the modification you made allowed `user1` to send emails successfully but not `user2`.

To resolve this issue and allow multiple users to send emails using their respective credentials, you should format the `sasl_passwd` file to associate each user with the relay host and port. Here's how you can adjust your `sasl_passwd`:

```plaintext
[xxx.smtp-relay.com]:465 user1@xxx.fit:smtp_relay_password1
[xxx.smtp-relay.com]:465 user2@xxx.fit:smtp_relay_password2
```

This format ensures that each user's credentials are correctly associated with the SMTP relay service on the specified port. Make sure to regenerate the hash database for Postfix after updating the `sasl_passwd` file by running:

```bash
postmap /tmp/docker-mailserver/sasl_passwd
```

Additionally, ensure that your `postfix-main.cf` and other configuration files are correctly referencing these files and that there are no typos or formatting issues.

If you continue to experience issues, it might be helpful to enable more detailed logging to get more insights into what might be going wrong. You can adjust the logging level in Postfix for more detailed output, which could provide further clues.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by CBAnother** [comment, marked as answer]
I solved it myself, `smtp_sender_dependent_authentication` needs to be set to yes.

**Reply by ashwaqrehman** [reply]
@CBAnother Can you check [this discussion](https://github.com/orgs/docker-mailserver/discussions/4031) , if you can help me on this
